src/bucket/BucketSnapshot.cpp

```diff
@@ -87,8 +87,10 @@ BucketSnapshot::getBucketEntry(LedgerKey const& k) const
 // do not load shadowed entries. If we don't find the entry, we do not remove it
 // from keys so that it will be searched for again at a lower level.
 void
-BucketSnapshot::loadKeys(std::set<LedgerKey, LedgerEntryIdCmp>& keys,
-                         std::vector<LedgerEntry>& result) const
+BucketSnapshot::loadKeysWithLimits(
+    std::set<LedgerKey, LedgerEntryIdCmp>& keys,
+    std::vector<LedgerEntry>& result,
+    std::shared_ptr<LedgerKeyMeter> lkMeter) const
```

Comment:
> (which I guess can be avoided as well by passing the reference to the whole XDR struct instead of just its size to canLoad).

I had originally just passed size as there was an optimization (not included in this PR) to partially deserialize the entry at the XDR stream level (to get the size and key of the entry) and check that its within quota. As things are implemented in this PR, I can change the interface to take the reference in this PR but it will be changed by a subsequent PR, so I don't think its worth it. 

However, I'll change the `shared_ptr` to an optional.

Reply:
Ok, that's really a nitpick. Feel free to skip if you plan to refactor this soon.